src/toil/jobStores/fileJobStore.py

```diff
@@ -351,6 +356,9 @@ def _writeToUrl(cls, readable, url):
         """
         # we use a ~10Mb buffer to improve speed
         atomic_copyobj(readable, cls._extractPathFromUrl(url), length=cls.BUFFER_SIZE)
+        if executable:
+            os.chmod(url.path, os.stat(url.path).st_mode | stat.S_IXUSR)
```

Comment:
Consider putting this functionality into `atomic_copyobj` and `atomic_copy`.  I'd intuitively expect a copy function to know how to preserve permissions while copying.